### DISCUSSION TITLE
write to i2s and copy to a2dp

### DISCUSSION THREAD

**Discussion by habuenav** [pschatzmann/arduino-audio-tools#1240] [Q&A] [answered]
Hello.
The situation is the following, I am writing a small and simple wav sample synthesizer, currently it is almost finished and generates the sound as I expect from the i2s output, I would also like to be able to send the sound to a bluetooth speaker, I have seen the examples streams -i2s -a2dp and base -i2s-a2dp and in both they treat the i2s as input (e.g. from a mic) and then copy to the a2dp, but in my case I have a function that writes the data to the i2s and I would like a2dp It will take what is arriving there and send it to a speaker, I have tried to copy i2s to i2s and from there to a2dp, but it only restarts the esp32, I appreciate any help. Thanks in advance

**Comment by pschatzmann** [comment, marked as answer]
I am not sure if I understand you correctly, but I have the impression that you are looking for the MultiOutput class ?

**Reply by habuenav** [reply]
Thanks, I'm working on it.
